lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -136,6 +128,54 @@ def build_client(options)
       Elasticsearch::Client.new(client_options)
     end
 
+    HOSTNAME_PORT_REGEX=/\A(?<hostname>([A-Za-z0-9\.\-]+))(:(?<port>\d+))?\Z/
```

Comment:
I'm not sure it's important, but this will fail to match IPv6 address:port combos which usually show up as `[ipv6]:port` (with the brackets, like localhost:9200 ipv6 would be `[::1]:9200`

We can just make a note about this and move on. I'm not sure many folks will need that anyway when DNS is usually simpler to work with ;P